python/tests/unit/processes/local_runtime/test_local_step.py

```diff
@@ -0,0 +1,558 @@
+
+
+async def test_invoke_function_calls_kernel_invoke():
+    """Test invoke_function calls the kernel's invoke method with provided arguments."""
+    mock_kernel = AsyncMock(spec=Kernel)
+    mock_kernel.invoke = AsyncMock()
+
+    mock_function = AsyncMock(spec=KernelFunction)
+
+    mock_step_info = AsyncMock(spec=KernelProcessStepInfo)
+    mock_step_info.edges = MagicMock(return_value={"edge1": "value1", "edge2": "value2"})
+    mock_step_info.state = KernelProcessStepState(name="test", id="step-id", state={})
+
+    step = LocalStep(
+        kernel=mock_kernel,  # type: ignore
+        step_info=mock_step_info,
+        outgoing_event_queue=Queue(),
+        event_namespace="ns",
+        step_state=mock_step_info.state,
+        factories={},
+    )
+
+    args = {"key": "value"}
+    await step.invoke_function(mock_function, mock_kernel, args)
+
+    mock_kernel.invoke.assert_called_once()
+
+    mock_kernel.invoke.assert_awaited_once_with(mock_function, **args)
+
+
+async def test_emit_event_puts_local_event_into_queue():
+    """Test emit_event creates a LocalEvent and puts it into the outgoing_event_queue."""
+    queue_obj = Queue()
+    mock_step_info = AsyncMock(spec=KernelProcessStepInfo)
+    mock_step_info.edges = MagicMock(return_value={"edge1": "value1", "edge2": "value2"})
+    mock_step_info.state = KernelProcessStepState(name="test", id="step-id", state={})
+
+    step = LocalStep(
+        kernel=AsyncMock(spec=Kernel),
+        step_info=mock_step_info,
+        outgoing_event_queue=queue_obj,
+        event_namespace="ns",
+        step_state=mock_step_info.state,
+        factories={},
+    )
+
+    event = KernelProcessEvent(id="test_event", data="some_data")
+    await step.emit_event(event)
+    # The queue should contain a LocalEvent
+    assert not queue_obj.empty()
+    local_event = queue_obj.get()
+    assert isinstance(local_event, LocalEvent)
+    assert local_event.inner_event is event
+    assert local_event.namespace == "test_step-id"
+
+
+async def test_emit_local_event_puts_into_queue():
+    """Test emit_local_event directly places the local_event into the queue with updated namespace."""
+    queue_obj = Queue()
+    mock_step_info = AsyncMock(spec=KernelProcessStepInfo)
+    mock_step_info.edges = MagicMock(return_value={"edge1": "value1", "edge2": "value2"})
+    mock_step_info.state = KernelProcessStepState(name="test", id="step-id", state={})
+
+    step = LocalStep(
+        kernel=AsyncMock(spec=Kernel),
+        step_info=mock_step_info,
+        outgoing_event_queue=queue_obj,
+        event_namespace="original_ns",
+        step_state=mock_step_info.state,
+        factories={},
+    )
+
+    local_event = LocalEvent(namespace="temp", inner_event=KernelProcessEvent(id="evt"))
+    await step.emit_local_event(local_event)
+
+    assert not queue_obj.empty()
+    popped = queue_obj.get()
+    # The namespace is updated by scoped_event
+    assert popped.namespace == f"{step.name}_{step.id}"
```

Comment:
We can also verify that the original `local_event` object is what gets queued (identity equality), to make sure there is no hidden copy.